### PR TITLE
Remove third-party vendors from cloud provider docs

### PR DESCRIPTION
**Pull request by MorrisLaw**
This PR updates the documentation on the kubernetes website: [k8s.io/docs/concepts/cluster-administration/cloud-providers/](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/). So that it aligns with the [community guidelines](https://kubernetes.io/docs/contribute/style/content-guide/#third-party-content) which prohibit third party content not necessary to the functionality of upstream Kubernetes. 

I used the list of `legacy` (previously known as `in-tree`) cloud providers as guidance on what should or shouldn't stay: [legacy cloud providers list](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/providers.go).

Removed these specific third-party vendors from the cloud provider docs:
- CloudStack
- Huawei Cloud (keeping link to project readme, since it's dual sourced and in-project)
- OVirt
- Photon
- IBM cloud
- Baidu cloud
- Tencent
- Alibaba

### CODE REVIEW

**Comment by MorrisLaw** [comment]
@zacharysarah here's that PR, for when you have time to review. I added a description including a link to what seems like _should_ stay. Let me know if this is correct. Thank you!

**Comment by kbhawkey** [comment]
Hello @MorrisLaw . Thanks for opening this pull request.

Page preview:
https://deploy-preview-21284--kubernetes-io-master-staging.netlify.app/docs/concepts/cluster-administration/cloud-providers/

**Comment by MorrisLaw** [comment]
> Hello @MorrisLaw . Thanks for opening this pull request.
> 
> Page preview:
> https://deploy-preview-21284--kubernetes-io-master-staging.netlify.app/docs/concepts/cluster-administration/cloud-providers/

Ahh thanks for that @kbhawkey . Looks like there's some leftover `Alibaba` documentation to remove. I'll update.

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 1]
```diff
@@ -134,15 +125,6 @@ If you wish to use the external cloud provider, its repository is [kubernetes/cl
 The GCE cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
 Note that the first segment of the Kubernetes Node name must match the GCE instance name (e.g. a Node named `kubernetes-node-2.c.my-proj.internal` must correspond to an instance named `kubernetes-node-2`).
 
-## HUAWEI CLOUD
-
-If you wish to use the external cloud provider, its repository is [kubernetes-sigs/cloud-provider-huaweicloud](https://github.com/kubernetes-sigs/cloud-provider-huaweicloud).
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md:139]: L137-139 Content in project; keep as is

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 2]
```diff
@@ -390,47 +358,4 @@ If you are running vSphere < 6.7U3, the in-tree vSphere cloud provider is recomm
 {{% /tab %}}
 {{< /tabs >}}
 
-For in-depth documentation on the vSphere cloud provider, visit the [vSphere cloud provider docs site](https://cloud-provider-vsphere.sigs.k8s.io).
-
-## IBM Cloud Kubernetes Service
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md:395]: @bradtopol 👋  This content meets the criteria for removal. Would you like to re-home it? Please let me know within 48 hours.

**Comment by MorrisLaw** [comment]
@jimangel hey, yes I will be able to address it within the next few days. Had a pretty busy sprint so I didn't get to it yet. But I will 😄

**Comment by MorrisLaw** [comment]
sorry for the delay, how does this look? @zacharysarah @sftim

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ weight: 30
 
 <!-- overview -->
 This page explains how to manage Kubernetes running on a specific
-cloud provider.
+cloud provider. There are many other third-party cloud provider projects, but this list is specific to projects embedded within, or relied upon by Kubernetes itself.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md:9]: At some point we should provide an overview of what cloud provider integration is (revising this page along with https://kubernetes.io/docs/concepts/architecture/cloud-controller/)

Comment (MorrisLaw) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md:9]: Yeah, I agree! It'd be helpful.

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md:139]: If L137-139 are kept, line 141-145 should be kept as well?

**Comment by MorrisLaw** [comment]
Are we waiting until July 16th to merge this @sftim ?

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md:395]: @bradtopol , Did you get a chance to review this change?

**Comment by kbhawkey** [comment]
Hi @MorrisLaw .
Would you have time to rebase this pull request?

**Comment by MorrisLaw** [comment]
@sftim @zacharysarah @kbhawkey @jimangel feedback has been addressed and this PR has been rebased.

**Pull request merged**